src/simsopt/geo/boozersurface.py

```diff
@@ -533,15 +542,67 @@ def solve_residual_equation_exactly_newton(self, tol=1e-10, maxiter=10, iota=0.,
             iota = x[-2]
             G = x[-1]
             i += 1
-            r, J = boozer_surface_residual(s, iota, G, self.biotsavart, derivatives=1)
+            r, J = boozer_surface_residual(s, iota, G, self.bs, derivatives=1)
 
-        res = {"residual": r, "jacobian": J, "iter": i, "success": norm <= tol,
-               "G": G, "s": s, "iota": iota}
-        return res
+        if s.stellsym:
+            J = np.vstack((
+                J[mask, :],
+                np.concatenate((label.dJ_by_dsurfacecoefficients(), [0., 0.])),
+            ))
+        else:
+            J = np.vstack((
+                J[mask, :],
+                np.concatenate((label.dJ_by_dsurfacecoefficients(), [0., 0.])),
+                np.concatenate((s.dgamma_by_dcoeff()[0, 0, 2, :], [0., 0.]))
+            ))
 
+        P, L, U = lu(J)
+        res = {
+            "residual": r, "jacobian": J, "iter": i, "success": norm <= tol, "G": G, "s": s, "iota": iota, "PLU": (P, L, U),
+            "mask": mask, 'type': 'exact'
+        }
+        self.res = res
+        self.need_to_run_code = False
+        return res
+   
+    def as_dict(self) -> dict:
+        d = {}
+        d["@class"] = self.__class__.__name__
+        d["@module"] = self.__class__.__module__
+        d["bs"] = self.bs
+        d["surface"] = self.surface
+        
+        # this needs to be rewritten so that labels are MSONable.  The problem
```

Comment:
NOTE: this is an temporary implementation of serialization for BoozerSurfaces.  The problem is described below, but I simply use `d['label']=self.label`, then when the BoozerSurface is deserialized, its label will be initialized with a `surface` object that is distinct from the `surface` object used by the BoozerSurface, even though they should be the _same_ surface objects.  I've discussed with @mbkumar and he's working on a fix for this issue.